packages/manager/src/features/CloudPulse/Alerts/Utils/utils.ts

```diff
@@ -626,3 +633,68 @@ export const transformDimensionValue = (
     )?.(value) ?? TRANSFORMS.capitalize(value)
   );
 };
+
+/**
+ * Invalidates the alerts cache
+ * @param qc The query client
+ * @param serviceType The service type
+ * @param entityId The entity id
+ * @param payload The payload
+ */
+export const invalidateAlerts = (
+  qc: QueryClient,
+  serviceType: string,
+  entityId: string,
+  payload: CloudPulseAlertsPayload
+) => {
+  const allAlerts = qc.getQueryData<Alert[]>(
+    queryFactory.alerts._ctx.alertsByServiceType(serviceType).queryKey
+  );
+
+  // Get alerts previously enabled for this entity
+  const oldEnabledAlertIds =
+    allAlerts
+      ?.filter((alert) => alert.entity_ids.includes(entityId))
+      .map((alert) => alert.id) || [];
+
+  // Combine enabled user and system alert IDs from payload
+  const newEnabledAlertIds = [
+    ...(payload.user ?? []),
+    ...(payload.system ?? []),
+  ];
+
+  // Get unique list of all enabled alert IDs for cache invalidation
+  const alertIdsToInvalidate = Array.from(
+    new Set([...oldEnabledAlertIds, ...newEnabledAlertIds])
+  );
```

Comment:
```suggestion
  const alertIdsToInvalidate = [...oldEnabledAlertIds, ...newEnabledAlertIds];
```
in worst case it will execute 2*size of alerts times, its okay it'll not make much difference